frontend/dockerfile/builder/build.go

```diff
@@ -54,6 +54,7 @@ const (
 
 var httpPrefix = regexp.MustCompile(`^https?://`)
 var gitUrlPathWithFragmentSuffix = regexp.MustCompile(`\.git(?:#.+)?$`)
+var keyPrefix = []string{"BUILDPLATFORM", "TARGETPLATFORM"} // add other args if needed.
```

Comment:
The builtin ones could be implicitly handled inside `dockerfile2llb` and you can define an extra `opts` key for the manual ones. For the implicit ones the logic could be that they are added if `PrefixPlatform` is set or `targetplatform!=buildplatform`. And add the other variants as well: `*OS` `*ARCH` `*VARIANT`.

Reply:
Do you mean show all those builtin args in the console?  is that too verbose? if we go this way it might looks like this, 
` => [linux/amd64 build 2/5] RUN echo "" TARGETPLATFORM=x TARGETOS =x, BUILDARCH=x, BUILDVARIANT=x ...`

and show all the args from `dispatchState.buildArgs` is enough.

Otherwise, the change here make it possible to show only partial of builtin args only if they are required to.

Just want to double confirm with your thoughts, thanks!

Reply:
or find a place to show all those value only once instead of each "RUN"?

Reply:
> Do you mean show all those builtin args in the console? is that too verbose? if we go this way it might looks like this,

Reply:
> Do you mean show all those builtin args in the console? is that too verbose? if we go this way it might looks like this

Only the args that have also been enabled with `ARG` would be shown. User probably wouldn't use `TARGETOS` and `TARGETARCH` together in same stage.